File: Semester 5/Mobile Application/Lab2/src/main/java/com/example/planner/ToDo.kt
package com.example.planner

import android.os.Parcel
import android.os.Parcelable

data class ToDo (
    val id: Long, //unique id for each activity
    val title: String,
    val time: String,
    val details: String,
    val goal: String,
    val emotion: String,
    var isChecked: Boolean = false
) : Parcelable {
    constructor(parcel: Parcel) : this (
        parcel.readLong(),
        parcel.readString() ?: "",
        parcel.readString() ?: "",
        parcel.readString() ?: "",
        parcel.readString() ?: "",
        parcel.readString() ?: "",
        parcel.readBoolean()
    )

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeLong(id)
        parcel.writeString(title)
        parcel.writeString(time)
        parcel.writeString(details)
        parcel.writeString(goal)
        parcel.writeString(emotion)
        parcel.writeBoolean(isChecked)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<ToDo> {
        override fun createFromParcel(parcel: Parcel): ToDo {
            return ToDo(parcel)
        }

        override fun newArray(size: Int): Array<ToDo?> {
            return arrayOfNulls(size)
        }
    }
}

File: Semester 5/Mobile Application/Lab3/lib/todo.dart
class Todo {
  int id;
  String title;
  String time;
  String details;
  String goal;
  String emotion;

  Todo({
    required this.id,
    required this.title,
    required this.time,
    required this.details,
    required this.goal,
    required this.emotion,
  });

  Todo copyWith({
    int? id,
    String? title,
    String? time,
    String? details,
    String? goal,
    String? emotion,
  }) {
    return Todo(
      id: id ?? this.id,
      title: title ?? this.title,
      time: time ?? this.time,
      details: details ?? this.details,
      goal: goal ?? this.goal,
      emotion: emotion ?? this.emotion,
    );
  }

  @override
  String toString() {
    return 'Todo(id: $id, title: $title, time: $time, details: $details, goal: $goal, emotion: $emotion)';
  }
}